dotnet3.5/STS/AssumeRole/AssumeRoleExampleTest/AssumeRoleTests.cs

```diff
@@ -0,0 +1,100 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+using Amazon;
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+using Moq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace AssumeRoleExampleTest
+{
+    public class AssumeRoleTests
+    {
+        private string roleArnToAssume = "arn:aws:iam::123456789012:role/testAssumeRole";
+        private static readonly RegionEndpoint REGION = RegionEndpoint.USWest2;
+
+        [Fact]
+        public async Task GetCallerIdentityAsyncTest()
+        {
+            // Create the mock SecurityToken client.
+            // Define variable
+            var mockClient = new Mock<AmazonSecurityTokenServiceClient>(REGION);
+            mockClient.Setup(client => client.GetCallerIdentityAsync(
+                    It.IsAny<GetCallerIdentityRequest>(),
+                    It.IsAny<CancellationToken>()))
+                .Callback<GetCallerIdentityRequest,
+                    CancellationToken>((request, token) =>
+                    {
+                    })
+                .Returns((GetCallerIdentityRequest r,
+                    CancellationToken token) =>
+                {
+                    return Task.FromResult(new GetCallerIdentityResponse()
+                    {
+                        Arn = roleArnToAssume,
+                        HttpStatusCode = HttpStatusCode.OK,
+                    });
+                });
+
+            var client = mockClient.Object;
+            var callerIdRequest = new GetCallerIdentityRequest();
+            var response = await client.GetCallerIdentityAsync(callerIdRequest);
+
+            bool ok = response.HttpStatusCode == HttpStatusCode.OK;
+            Assert.True(ok, $"Successfully retrieved caller Identity.");
+
+        }
+
+        [Fact]
+        public async Task AssumeRoleAsyncTest()
+        {
+            // Create the mock client object
+            var mockClient = new Mock<AmazonSecurityTokenServiceClient>(REGION);
+            mockClient.Setup(client => client.AssumeRoleAsync(
+                    It.IsAny<AssumeRoleRequest>(),
+                    It.IsAny<CancellationToken>()))
+                .Callback<AssumeRoleRequest,
+                    CancellationToken>((request, token) =>
+                    {
+                        if (request is not null)
+                        {
+
+                        }
```

Comment:
Did you mean to put something in this if block?

Reply:
No. Removed.